source/common/network/socket_impl.h

```diff
@@ -93,6 +98,7 @@ class ConnectionInfoSetterImpl : public ConnectionInfoSetter {
   std::string ja3_hash_;
   absl::optional<std::chrono::milliseconds> round_trip_time_;
   FilterChainInfoConstSharedPtr filter_chain_info_;
+  OptRef<const ListenerInfo> listener_info_;
```

Comment:
hmmm, the premise of my previous comments is that connection info provider will own a copy of listener info shared pointer.

So, I think the listenerInfo() of ListenerConfig need to change it's signature. But seems if we don't change the signature, there is no way to store a listener info shared pointer to the connection info provider.

Seems I am wrong, my bad, again.

Reply:
Maybe revert to your previous implementation would be better (use a shared pointer here).

Sorry again. 😞

Reply:
No problem, reverted. It was worth exploring the other option.